File: programmers/lv2/튜플.py
def solution(s):
    answer = []
    
    tuples = arr = []
    num, check = '', False
    for char in s[1:-1]:
        if char == '{':
            arr, check = [], True
            continue
        elif char == '}':
            tuples.append(arr + [int(num)])
            arr, num, check = [], '', False
        elif char == ',':
            if check:
                arr.append(int(num))
            num = ''
        else:
            num += char
    tuples.sort(key=lambda x:len(x))
    
    tup_dict = {}
    for arr in tuples:
        for num in arr:
            if not num in tup_dict:
                tup_dict[num] = num
                answer.append(num)
    
    return answer

File: programmers/lv2/기출문제/튜플.js
function solution(s) {
    var answer = [];
    
    var num = 0;
    const map = new Map();
    for(var c = 0; c < s.length; c++) {
        const char = s.charAt(c);
        if(!isNaN(char)) {
            num = num * 10 + parseInt(char);
            continue;
        }
        if(char === '}' || char === ',') {
            if(num !== 0) {
                if(map.has(num)) {
                    map.set(num, map.get(num) + 1);
                } else {
                    map.set(num, 1);
                }
            }
        }
        num = 0;
    }
    
    const arr = [];
    for(const [key, value] of map.entries()) {
        arr.push([key, value]);
    }
    
    arr.sort((a, b) => b[1] - a[1]);
    
    answer = arr.map(item => item[0]);
    
    return answer;
}
